### ISSUE TITLE
Error occured when serverpod hosting to render using docker via github repo

### ISSUE THREAD

**Issue by brandscart** [serverpod/serverpod#2542]
Pushing image to registry...
Upload succeeded
DONE
==> Deploying...
Failed to load passwords. Serverpod cannot not start.
==> Exited with status 1

i have faced above issue when deploying the docker container to render.Please let me know solution if any one know?
Thanks in advance.

i have pasted my dockerfile below for reference.

# If you update the dart version, make sure the image is
# compatible with the busybox image.
FROM dart:3.2.5 AS build

WORKDIR /app
COPY . .

RUN dart pub get
RUN dart compile exe bin/main.dart -o bin/main

# If you update the busybox version, make sure the image is
# compatible with the dart image.
FROM debian:stable
RUN apt-get update && apt-get install -y libc6

ENV runmode=production
ENV serverid=default
ENV logging=normal
ENV role=monolith

COPY --from=build /app/bin/main /app/bin/main
COPY --from=build /app/config/ config/
COPY --from=build /app/web/ web/

EXPOSE 8080
EXPOSE 8081
EXPOSE 8082

ENV SERVERPOD_PASSWORDS_PATH=/app/config/passwords.yaml

CMD app/bin/main --mode $runmode --server-id $serverid --logging $logging --role $role

**Comment by Isakdl** [comment]
Hey @brandscart 

The issue you get is that the `passwords.yaml` file cannot be found, my guess is that you do not have the passwords.yaml file inside the config folder when you are building the container.

**Comment by brandscart** [comment]
> Hey @brandscart
> 
> The issue you get is that the `passwords.yaml` file cannot be found, my guess is that you do not have the passwords.yaml file inside the config folder when you are building the container.

Yes, i forgot to add password.yaml to my private repo

**Comment by brandscart** [comment]
Hi @Isakdl 

I have face another issue in deploying. 

=> Deploying...
SERVERPOD version: 1.2.5, dart: 3.2.5 (stable) (Tue Jan 16 15:02:13 2024 +0000) on "linux_x64", time: 2024-07-30 16:51:26.295601Z
mode: production, role: monolith, logging: normal, serverId: default
Failed to connect to the database. Retrying in 10 seconds. TimeoutException after 0:00:15.000000: Future not completed
===== asynchronous gap ===========================
package:postgres_pool/postgres_pool.dart 351                PgPool.run.<fn>
package:retry/retry.dart 131                                RetryOptions.retry
package:postgres_pool/postgres_pool.dart 349                PgPool.run
package:serverpod/src/database/database_connection.dart 42  DatabaseConnection.testConnection
package:serverpod/src/server/serverpod.dart 673             Serverpod._connectToDatabase
package:serverpod/src/server/serverpod.dart 354             Serverpod.start.<fn>
package:serverpod/src/server/serverpod.dart 341             Serverpod.start
package:brandscart_server/server.dart 43                    run
Database configuration:
database host: database.brandscart.github.io
database port: 5432
database name: serverpod
database user: postgres
database require SSL: false
database unix socket: false
database pass: ********
==> No open ports detected, continuing to scan...
==> Docs on specifying a port: https://render.com/docs/web-services#port-binding


why i am get this error in the last time?. Please let me know the reason.

**Comment by Isakdl** [comment]
Your serverpod instance is unable to connect to the database. This is a configuration issue in your environment.

Is the database running? Are the connection details correct? Do you have network access to the db? Are you sure you shouldn't use SSL? Your config has set `database require SSL: false`.

**Comment by brandscart** [comment]
> Your serverpod instance is unable to connect to the database. This is a configuration issue in your environment.
> 
> Is the database running? Are the connection details correct? Do you have network access to the db? Are you sure you shouldn't use SSL? Your config has set `database require SSL: false`.

@Isakdl 

Yes it is connection issue to database in production, connecting attempt failed in production config. But in local i can connect to my postgres database easily.


iam noobie in backend development, so i don't know well how to live our server .can you suggest any resources regarding deploying serverpod to render?

**Comment by Isakdl** [comment]
You have to set up a database in your production environment and put in all the credentials/connection info in the configuration file. It looks like you are using the template URL "database.private-production.examplepod.com", this is not a real URL. You'll need to replace this with the one too your db. 

Consult the documentation on your cloud provider for how to create a new database and where you can find the credentials / connection info.
